src/Symfony/Component/Runtime/SymfonyRuntime.php

```diff
@@ -0,0 +1,249 @@
+
+            return $this->start($application);
+        }
+
+        if ($app instanceof Application) {
+            if (!\in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true)) {
+                echo 'Warning: The console should be invoked via the CLI version of PHP, not the '.\PHP_SAPI.' SAPI'.PHP_EOL;
+            }
+
+            set_time_limit(0);
+            $input = $this->getInput();
+            $env = $this->env;
+
+            return static function () use ($app, $input, $env): int {
+                if (null !== $env) {
+                    return $app->run($input);
+                }
+
+                $definition = $app->getDefinition();
+
+                if (!$definition->hasOption('env') && !$definition->hasOption('e') && !$definition->hasShortcut('e')) {
+                    $definition->addOption(new InputOption('--env', '-e', InputOption::VALUE_REQUIRED, 'The Environment name.', $_SERVER['APP_ENV'] ?? 'dev'));
+                }
+
+                if (!$definition->hasOption('no-debug')) {
+                    $definition->addOption(new InputOption('--no-debug', null, InputOption::VALUE_NONE, 'Switches off debug mode.'));
+                }
+
+                return $app->run($input);
+            };
+        }
+
+        return parent::start($app);
+    }
+
+    protected function getArgument(\ReflectionParameter $parameter, ?\ReflectionType $type)
+    {
+        if (!$type instanceof \ReflectionNamedType) {
+            return parent::getArgument($parameter, $type);
+        }
+
+        switch ($type->getName()) {
+            case Request::class:
+                return $this->request ?? $this->request = Request::createFromGlobals();
+
+            case InputInterface::class:
+                return $this->getInput();
+
+            case OutputInterface::class:
+                return $this->output ?? $this->output = new ConsoleOutput();
+
+            case Application::class:
+                return $this->application ?? $this->application = new Application();
+
+            case Command::class:
+                return $this->command ?? $this->command = new Command();
+        }
+
+        return parent::getArgument($parameter, $type);
+    }
+
+    private function getInput(): ArgvInput
+    {
+        if (null !== $this->input) {
+            return $this->input;
+        }
+
+        $input = new ArgvInput();
+
+        if (null !== $this->env) {
+            return $this->input = $input;
+        }
+
+        if (null !== $env = $input->getParameterOption(['--env', '-e'], null, true)) {
+            putenv('APP_ENV='.$_SERVER['APP_ENV'] = $_ENV['APP_ENV'] = $env);
+        }
+
+        if ($input->hasParameterOption('--no-debug', true)) {
+            putenv('APP_DEBUG='.$_SERVER['APP_DEBUG'] = $_ENV['APP_DEBUG'] = '0');
+        }
+
+        return $this->input = $input;
+    }
+
+    private function wrapCommand(\Closure $app): \Closure
+    {
+        return function () use ($app) {
```

Comment:
Same here, we could use a class to
1. ease reading.
2. move the logic there

Reply:
Same as above, I fail to see how the boilerplate of a class is going to ease reading, compared to an inline closure...